data-diff-demo/data_diff_demo/healing_diff_demo.py

```diff
@@ -135,30 +89,74 @@ def data_diff_healing_check() -> AssetCheckResult:
 
     total_healed_diffs_count = len(healed_diff_results)
 
-    yield AssetCheckResult(
-        passed=total_healed_diffs_count == 0,
-        severity=AssetCheckSeverity.ERROR,
-        metadata={
-            "total_diffs_count_original": MetadataValue.int(total_diffs_count),
-            "total_diffs_unhealed": MetadataValue.int(total_healed_diffs_count),
-            "source_row_diffs": MetadataValue.int(
-                len(results[results["diff_type"] == "-"])
-            ),
-            "target_row_diffs": MetadataValue.int(
-                len(results[results["diff_type"] == "+"])
-            ),
-            "preview_all_diffs": MetadataValue.md(results.head(100).to_markdown()),
-            "preview_diff_overwrites": MetadataValue.md(
-                source_row_diffs.head(100).to_markdown()
-            ),
-            "preview_diffs_remaining": MetadataValue.md(
-                healed_diff_results.head(100).to_markdown()
-            ),
-        },
+    context.log_event(
+        AssetObservation(
+            asset_key="replicated_healing_events",
+            metadata={
+                "total_diffs_count_original": MetadataValue.int(total_diffs_count),
+                "total_diffs_unhealed": MetadataValue.int(total_healed_diffs_count),
+                "source_row_diffs": MetadataValue.int(
+                    len(results[results["diff_type"] == "-"])
+                ),
+                "target_row_diffs": MetadataValue.int(
+                    len(results[results["diff_type"] == "+"])
+                ),
+                "preview_all_diffs": MetadataValue.md(results.head(100).to_markdown()),
+                "preview_diff_overwrites": MetadataValue.md(
+                    source_row_diffs.head(100).to_markdown()
+                ),
+                "preview_diffs_remaining": MetadataValue.md(
+                    healed_diff_results.head(100).to_markdown()
+                ),
+            },
+        )
     )
 
+    if total_healed_diffs_count == 0:
+        context.log.info("All diffs healed!")
+        return total_healed_diffs_count
+    else:
+        raise Exception("Diffs remain after healing!")
+
+
+@graph_asset()
+def replicated_healing_events(source_healing_events):
+    src_conn = duckdb.connect(SOURCE_DATABASE_PATH)
```

Comment:
The logic between L124 and L155 would benefit from getting moved into separate ops, thus having the `@graph_asset`-decorated method being solely responsible for executing the ops and not any computation in itself.

Reply:
@tacastillo updated. How does it look now?